common/dynamicconfig/file_based_client.go

```diff
@@ -177,9 +198,17 @@ func (fc *fileBasedClient) Update() error {
 
 	prev := fc.values.Swap(newValues)
 	oldValues, _ := prev.(configValueMap)
-	fc.logDiff(oldValues, newValues)
+	changedMap := fc.diffAndLog(oldValues, newValues)
```

Comment:
Is it worth exiting early if the changedMap is empty?

Reply:
sure